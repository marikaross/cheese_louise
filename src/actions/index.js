export const addCheese = (cheeses) => ({
    type: 'ADD_CHEESE',
    cheeses
})

export const toggleFaveCheese = (cheeseId) => ({
    type: 'TOGGLE_FAVORITE_CHEESE',
    cheeseId
})

export const addFaveCheese = (id) => ({
  type: 'ADD_FAVE_CHEESE',
  id
})

export const deleteCheese = (id) => ({
  type: 'DELETE_CHEESE',
  id
})

export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})

export const hasErrored = (bool) => ({
  type: 'HAS_ERRORED',
  hasErrored: bool
})

export const cheeseFetchDataSuccess = (cheeses) => ({
  type: 'CHEESE_FETCH_DATA_SUCCESS',
  cheeses
})

export const summaryFetchDataSuccess = (summary, id) => ({
  type: 'SUMMARY_FETCH_DATA_SUCCESS',
  id, 
  summary
})
