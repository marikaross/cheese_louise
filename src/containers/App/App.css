.App {
  text-align: center;
  font-family: 'Rokkitt', serif;
  font-family: 'Open Sans', sans-serif;
  background: #fefaea;
  width: 100vw;
  height: 1400px;
}


