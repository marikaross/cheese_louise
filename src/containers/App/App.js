import React, { Component } from 'react';
import CheeseContainer from '../CheeseContainer/CheeseContainer';
import Favorites from '../Favorites/Favorites';
import { Header } from '../Header/Header';
import { connect } from 'react-redux';
import { addCheese } from '../../actions';
import { fetchCheese } from '../../thunks/fetchCheese';
import { Route, withRouter } from 'react-router-dom';
import './App.css';



export class App extends Component{
componentDidMount() {
    const url = 'https://data.opendatasoft.com/api/records/1.0/search/?dataset=frenchcheese%40public&facet=id&facet=cheese&facet=milk'
    this.props.fetchCheese(url)
  }
  

  render() {
    return (
      <div className="App">
        <Route path='/' component={Header} />
        <Route exact path='/' component={CheeseContainer} />
        <Route exact path='/favorites' component={Favorites} />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  cheeses: this.state.cheese,
  favorites: this.state.favorites,
  isLoading: this.state.isLoading,
  hasErrored: this.state.hasErrored
})

export const mapDispatchToProps = (dispatch) => ({
  addCheese: (cheeses) => dispatch(addCheese(cheeses)), 
  fetchCheese: (url) => dispatch(fetchCheese(url))
})

export default withRouter(connect(mapDispatchToProps, mapDispatchToProps)(App))
