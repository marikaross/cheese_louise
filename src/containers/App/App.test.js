import {App, mapStateToProps, mapDispatchToProps} from './App';
import React from 'react';
import { shallow } from 'enzyme';
import * as action from '../../actions';

describe('App', () => {
  let wrapper
  let mockCheeses
  let mockFavorites
  let mockIsLoading
  let mockHasErrored
  let mockFetchCheese
  let mockAddCheese

  beforeEach(() => {
    mockCheeses = [{name: 'nacho', id: 6}, {name: 'swiss', id: 5}]
    mockFavorites = [{name: 'nacho', id: 6}]
    mockFetchCheese = jest.fn()
    mockAddCheese = jest.fn()
    mockHasErrored = false
    mockIsLoading = true

     wrapper = shallow(<App 
      cheeses={mockCheeses}
      favorites={mockFavorites}
      isLoading={mockIsLoading}
      hasErrored={mockHasErrored}
      addCheese={mockAddCheese}
      fetchCheese={mockFetchCheese}/>)
  })

  it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should return an object with a cheeses array and a favorites array', () => {
      const mockState = {
        cheeses: mockCheeses,
        favorites: mockFavorites,
        addCheese: mockAddCheese
      }

      const expectedProps = {
        cheeses: mockCheeses,
        favorites: mockFavorites
      }

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expectedProps)
  })

  it('should call dispatch with addCheese', () => {
      const mockState = {
        cheeses: mockCheeses,
        favorites: mockFavorites,
        addCheese: mockAddCheese,
        fetchCheese: mockFetchCheese
      }

      const mockDispatch = jest.fn()
      const mappedProps = mapDispatchToProps(mockDispatch)
      const actionToDispatch = action.addCheese({name: 'american'})
      mappedProps.addCheese({name: 'american'})
      expect(mockDispatch).toBeCalledWith(actionToDispatch)
  })

})