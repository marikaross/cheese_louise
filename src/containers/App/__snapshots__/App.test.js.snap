// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addCheese={[Function]}
    cheeses={
        Array [
            Object {
              "id": 6,
              "name": "nacho",
            },
            Object {
              "id": 5,
              "name": "swiss",
            },
          ]
    }
    favorites={
        Array [
            Object {
              "id": 6,
              "name": "nacho",
            },
          ]
    }
    fetchCheese={[Function]}
    hasErrored={false}
    isLoading={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          path="/"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/favorites"
/>,
        <Route
          exact={true}
          path="/:cheeseId"
          render={[Function]}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/favorites",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/:cheeseId",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/favorites"
/>,
          <Route
            exact={true}
            path="/:cheeseId"
            render={[Function]}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/favorites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/:cheeseId",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
